mage_ai/data_preparation/models/pipeline.py

```diff
@@ -729,6 +772,10 @@ def save(self, block_uuid: str = None, widget: bool = False):
                 current_pipeline.blocks_by_uuid[block_uuid] = block
             pipeline_dict = current_pipeline.to_dict()
         else:
+            if self.data_integration is not None:
+                with open(self.catalog_config_path, 'w') as fp:
+                    json.dump(self.data_integration, fp)
+                self.data_integration = None
```

Comment:
why do we set it to None? you can exclude it from the `to_dict()` method. setting it to None could cause problems.

Reply:
Got it. I was testing it to make sure there weren't any issues, but better to not set it to `None`. Was trying to figure out a way to exclude the data_integration prop from the dictionary without having to modify the `to_dict()` method.